src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,140 @@
+                        Task item = list.get(i);
+                        System.out.println(String.format("     %d. %s", (i + 1), item.toString()));
+                    }
+                }
+            } else if (parsedInput.equals("done")) {
+                if (input.replaceAll("\\s", "").length() == 4) {
+                    throw new DukeException(DukeException.Type.EmptyDone);
+                } else {
+                    String modInput = input.replaceAll("\\s", "");
+                    int index = Integer.parseInt(modInput.substring(modInput.length() - 1)) - 1;
+                    if (list.size() <= index) {
+                        throw new DukeException(DukeException.Type.InvalidTaskNumber);
+                    } else {
+                        Task item = list.get(index);
+                        item.markAsDone();
+                        System.out.println("    ____________________________________________________________");
+                        System.out.println("     Nice! I've marked this task as done:");
+                        System.out.println(String.format("       %s", item.toString()));
+                    }
+                }
+            } else if (parsedInput.equals("delete")) {
+                if (input.replaceAll("\\s", "").length() == 6) {
+                    throw new DukeException(DukeException.Type.EmptyDelete);
+                } else {
+                    String modInput = input.replaceAll("\\s", "");
+                    int index = Integer.parseInt(modInput.substring(modInput.length() - 1)) - 1;
+                    if (list.size() <= index) {
+                        throw new DukeException(DukeException.Type.InvalidTaskNumber);
+                    } else {
+                        Task item = list.remove(index);
+                        System.out.println("    ____________________________________________________________");
+                        System.out.println("     Noted. I've removed this task:");
+                        System.out.println(String.format("       %s", item.toString()));
+                        System.out.println(String.format("     Now you have %d tasks in the list.", list.size()));
+                    }
+                }
+            } else if (parsedInput.equals("todo")) {
+                if (input.replaceAll("\\s", "").length() == 4) {
+                    throw new DukeException(DukeException.Type.EmptyTodo);
+                } else {
+                    Todo t = new Todo(input.substring(5));
+                    System.out.println();
+                    list.add(t);
+                    System.out.println("    ____________________________________________________________");
+                    System.out.println("     Got it. I've added this task:");
+                    System.out.println("       " + t.toString());
+                    System.out.println(String.format("     Now you have %d tasks in the list.", list.size()));
```

Comment:
These lines of codes are used more than twice. Consider abstracting to a method to enhance ease of readability and maintainability.